### ISSUE TITLE
LightXML on other dataset

### ISSUE THREAD

**Issue by wangchichi1999** [kongds/LightXML#3]
Hi!
Did I need to do something for label clustering when I train LightXML on my own datasets other than [eurlex4k|wiki31k|amazon13k|amazon670k|wiki500k]
Why does LightXML only use label clustering on Wiki500k&670k?

**Comment by kongds** [comment]
Like [AttentionXML](https://github.com/yourh/AttentionXML), the purpose of label clustering is to process large datasets like Wiki500K&Amazon670K, not to improve the precision.  

So if your datasets have hundreds of thousands of labels, you may need to use label clustering.

**Comment by wangchichi1999** [comment]
Thanks a loooooooooooooooot!
Yes! I want to run it on a 500k_raw_text_dataset But I don't have .txt in this format(like train.txt in EUR-4K)
Could you tell me how can I transfer my text to this "number format"? 
Thanks a lot agaaaaaaaaain!
![RJ4C_60Z$%@NKRZ5ONQ}6FY](https://user-images.githubusercontent.com/62172616/105137362-1f7f6d00-5b2e-11eb-9bdf-46da3ec77ce8.png)
![HS~HO$5CXOZFITK@(WG EA5](https://user-images.githubusercontent.com/62172616/105137004-98ca9000-5b2d-11eb-837a-5ec1558a4720.png)

**Comment by kongds** [comment]
The format of train.txt is svmlight. You need to get tf-idf from raw_text, then use `sklearn.datasets.dump_svmlight_file` to dump tf-idf and labels.

**Comment by wangchichi1999** [comment]
Thanks for your generous help!

**Comment by Dave-AdamsWANG** [comment]
> 

Hi! For datasets other than these two huge datasets, do you just set group_y as None in class LightXML?

**Comment by kongds** [comment]
Yes
